File: 728/728.cpp
/*!***********************
* @brief:  leetcode-728-自除数
* @date:   2019/5/6
* @envir:  gcc 7.4.0
* @author: luhao
* ***********************/

class Solution {
public:
    vector<int> selfDividingNumbers(int left, int right) {
        vector<int> ans;
        for (int i=left; i<=right; i++){
            bool flag = 1;
            int x = i;
            while (x!=0){
                if (x%10==0 || i%(x%10)!=0){
                    flag = 0;
                    break;
                }
                x /= 10;
            }
            if (flag)
                ans.push_back(i);
        }
        return ans;
    }
};

File: 728/728.py
# @brief:  leetcode-728-自除数
# @envir:  python 2.7
# @date:   2019/5/6
# @author: luhao

class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        ans = []
        for i in range(left,right+1):
            x = i
            flag = True
            while x!=0:
                if x%10 == 0 or i % (x%10)!=0:
                    flag = False
                    break
                x /= 10
            if flag:
                ans.append(i)
        return ans
